src/sys/termios.rs

```diff
@@ -1,648 +1,782 @@
+            BaudRate::B1000000 => B1000000,
+            #[cfg(any(target_os = "android", target_os = "linux"))]
+            BaudRate::B1152000 => B1152000,
+            #[cfg(any(target_os = "android", target_os = "linux"))]
+            BaudRate::B1500000 => B1500000,
+            #[cfg(any(target_os = "android", target_os = "linux"))]
+            BaudRate::B2000000 => B2000000,
+            #[cfg(any(target_os = "android", target_os = "linux"))]
+            BaudRate::B2500000 => B2500000,
+            #[cfg(any(target_os = "android", target_os = "linux"))]
+            BaudRate::B3000000 => B3000000,
+            #[cfg(any(target_os = "android", target_os = "linux"))]
+            BaudRate::B3500000 => B3500000,
+            #[cfg(any(target_os = "android", target_os = "linux"))]
+            BaudRate::B4000000 => B4000000,
         }
     }
+}
 
-    #[cfg(any(target_os = "linux", target_os = "android"))]
-    pub mod consts {
-        use libc::{c_int, c_uint, c_uchar};
-
-        pub type tcflag_t = c_uint;
-        pub type cc_t = c_uchar;
-        pub type speed_t = c_uint;
-
-        #[repr(C)]
-        #[derive(Clone, Copy)]
-        pub struct Termios {
-            pub c_iflag: InputFlags,
-            pub c_oflag: OutputFlags,
-            pub c_cflag: ControlFlags,
-            pub c_lflag: LocalFlags,
-            pub c_line: cc_t,
-            pub c_cc: [cc_t; NCCS],
-            pub c_ispeed: speed_t,
-            pub c_ospeed: speed_t
-        }
+/// Specify when a port configuration change should occur.
+///
+/// Used as an argument to `tcsetattr()`
+// TODO: Add TCSASOFT, which will require treating this as a bitfield.
+#[derive(Clone, Copy, Debug)]
+pub enum SetArg {
+    /// The change will occur immediately
+    TCSANOW = libc::TCSANOW as isize,
+    /// The change occurs after all output has been written
+    TCSADRAIN = libc::TCSADRAIN as isize,
+    /// Same as `TCSADRAIN`, but will also flush the input buffer
+    TCSAFLUSH = libc::TCSAFLUSH as isize,
+}
 
-        #[derive(Clone, Copy, Debug, PartialEq)]
-        pub enum BaudRate {
-            B0,
-            B50,
-            B75,
-            B110,
-            B134,
-            B150,
-            B200,
-            B300,
-            B600,
-            B1200,
-            B1800,
-            B2400,
-            B4800,
-            B9600,
-            B19200,
-            B38400,
-            B57600,
-            B115200,
-            B230400,
-            B460800,
-            B500000,
-            B576000,
-            B921600,
-            B1000000,
-            B1152000,
-            B1500000,
-            B2000000,
-            B2500000,
-            B3000000,
-            B3500000,
-            B4000000,
-        }
+/// Specify a combination of the input and output buffers to flush
+///
+/// Used as an argument to `tcflush()`.
+#[derive(Clone, Copy, Debug)]
+// TODO: Change these over to a `libc_enum` macro
+pub enum FlushArg {
```

Comment:
as with `SetArg`, you should use `#[repr(i32)]` and ditch the `as isize`.  What's a `libc_enum` macro?